crates/uv/src/commands/workspace/run.rs

```diff
@@ -58,53 +56,112 @@ pub(crate) async fn run(
         "python".to_string()
     };
 
-    // Copy the requirements into a set of overrides; we'll use this to prioritize
-    // requested requirements over those discovered in the project.
-    // We must retain these requirements as direct dependencies too, as overrides
-    // cannot be applied to transitive dependencies.
-    let overrides = requirements.clone();
+    // Discover and sync the workspace.
+    let workspace_env = if isolated {
+        None
+    } else {
+        debug!("Syncing workspace environment.");
 
-    if !isolated {
-        if let Some(workspace_requirements) = workspace::find_workspace()? {
-            requirements.extend(workspace_requirements);
-        }
-    }
+        let Some(workspace_requirements) = workspace::find_workspace()? else {
+            return Err(anyhow::anyhow!(
+                "Unable to find `pyproject.toml` for project workspace."
+            ));
+        };
 
-    // Detect the current Python interpreter.
-    // TODO(zanieb): Create ephemeral environments
-    // TODO(zanieb): Accept `--python`
-    let run_env = environment_for_run(
-        &requirements,
-        &overrides,
-        python.as_deref(),
-        isolated,
-        preview,
-        cache,
-        printer,
-    )
-    .await?;
-    let python_env = run_env.python;
+        let venv = PythonEnvironment::from_virtualenv(cache)?;
 
-    // Construct the command
-    let mut process = Command::new(&command);
-    process.args(&args);
+        // Sync the workspace requirements.
+        Some(sync_environment(venv, &workspace_requirements, preview, cache, printer).await?)
```

Comment:
Just throwing this out here to think about... wouldn't it be annoying if we exposed a default environment at `.venv` and every time you used `uv run` any manual changes you had made were undone?

Reply:
Yeah, though this doesn't uninstall anything, it just makes sure it's in-sync with the (in-the-future) lockfile. Does that seem right?

Reply:
Ah we'll need to be careful with "in sync" because sync implies an exact match (no extra packages) to me. Seems ideal not to remove packages unless requested? Like.. `uv sync --strict`?

Reply:
This is in uv run, not uv sync. For now, neither one is removing packages, but I don’t feel strongly about the behavior.

Reply:
I understand that, I'm talking more generally.

`uv run` should not probably not remove extraneous packages (and I understand that is the current behavior).